<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1119</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Нейронная сеть</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>691</width>
      <height>571</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Основное окно</string>
     </attribute>
     <widget class="QTextEdit" name="textEdit">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>461</width>
        <height>191</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="iteration">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>10</y>
        <width>91</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="calc">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>40</y>
        <width>87</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Посчитать</string>
      </property>
     </widget>
     <widget class="QPushButton" name="clear_text">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>70</y>
        <width>87</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Очистить</string>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>10</y>
        <width>91</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>40</y>
        <width>87</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Подлючить</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>351</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>Параметры манипулятора</string>
      </property>
      <widget class="QLabel" name="label_19">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>L1</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_20">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>L2</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_21">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>L3</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_22">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>L4</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_23">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>L5</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="L1">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="L2">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="L3">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="L4">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="L5">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="setDefaultParams">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>40</y>
         <width>87</width>
         <height>27</height>
        </rect>
       </property>
       <property name="text">
        <string>Применить</string>
       </property>
      </widget>
     </widget>
     <widget class="QCheckBox" name="timer_check">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>110</y>
        <width>90</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>timer</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Прямая кинематика</string>
     </attribute>
     <widget class="QPushButton" name="pr_kin_calc">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>70</y>
        <width>87</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Рассчитать</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="pr_coordinate">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>111</width>
        <height>80</height>
       </rect>
      </property>
      <property name="acceptDrops">
       <bool>false</bool>
      </property>
      <property name="title">
       <string>Координаты</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>X</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Y</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Z</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="pr_x">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="pr_y">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="pr_z">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="pr_angle">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>261</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>Углы</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QLineEdit" name="pr_Q1">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q1</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="pr_Q2">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="pr_Q3">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="pr_Q4">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="pr_Q5">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>50</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q2</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q3</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q4</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q5</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="pr_random">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>30</y>
        <width>87</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Случайно</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="pr_with_input">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>100</y>
        <width>131</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string notr="true">С учетом новых
значений</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>20</y>
        <width>171</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>Обратная задача кинематики рассчитывается с помощью математической модели манипулятора робота</string>
      </property>
      <property name="scaledContents">
       <bool>false</bool>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Обратная кинематика</string>
     </attribute>
     <widget class="QGroupBox" name="pr_angle_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>281</width>
        <height>171</height>
       </rect>
      </property>
      <property name="title">
       <string>Углы</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QLineEdit" name="ob_Q1">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>50</y>
         <width>71</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>15</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q1</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="ob_Q2">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>50</y>
         <width>81</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>20</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="ob_Q3">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>50</y>
         <width>101</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>30</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="ob_Q4">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>120</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>24</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="ob_Q5">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>120</y>
         <width>41</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>17</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q2</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>30</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q3</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q4</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>100</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Q5</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="ob_random">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>30</y>
        <width>87</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Случайно</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="pr_coordinate_2">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>111</width>
        <height>80</height>
       </rect>
      </property>
      <property name="acceptDrops">
       <bool>false</bool>
      </property>
      <property name="title">
       <string>Координаты</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>X</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Y</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>20</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Z</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="ob_x">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>19</string>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="ob_y">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>20</string>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QLineEdit" name="ob_z">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>40</y>
         <width>31</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>5</string>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="ob_kin_calc">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>70</y>
        <width>87</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Рассчитать</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="ob_with_input">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>100</y>
        <width>131</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string notr="true">С учетом новых
значений</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>30</y>
        <width>171</width>
        <height>111</height>
       </rect>
      </property>
      <property name="text">
       <string>Обратная задача кинематики рассчитывается с помощью нейронной сети. Перед этим необходимо обучить сеть с помощью прямой задачи кинематики</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_24">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>350</y>
      <width>321</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Входные данные для нейрнной сети</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="nnInputValue">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>370</y>
      <width>341</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_25">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>470</y>
      <width>331</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Полученные данный нойронной сетью</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="nnOutputValue">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>490</y>
      <width>341</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="nnGetStudy">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>520</y>
      <width>87</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Обучить</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_26">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>420</y>
      <width>321</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Расчетные данные выхода нейронной сети</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="nnMathOutputValue">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>440</y>
      <width>341</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget_2">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>0</y>
      <width>381</width>
      <height>281</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>КСР</string>
     </attribute>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>355</width>
        <height>205</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>График обучение</string>
     </attribute>
     <widget class="QGraphicsView" name="graphicsNNTeach">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>355</width>
        <height>205</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1119</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
